pkg/rule/rule.go

```diff
@@ -629,6 +674,29 @@ func runBackgroundCommandOnPods(pods []v1.Pod, container, cmd, taskID, cmdExecut
 	}()
 
 	logrus.Infof("Created pod command executor: [%s] %s", createdPod.GetNamespace(), createdPod.GetName())
+	// Check whether the cmd executor pod is struck in pending state for more than five mintues
+	// If struck, delete the pod and return error.
+	for i := 0; i < maxRetry; i++ {
+		p, err := core.Instance().GetPodByUID(createdPod.GetUID(), createdPod.GetNamespace())
+		if err != nil {
+			return err
+		}
+		if p.Status.Phase == v1.PodPending {
+			if i == (maxRetry - 1) {
+				var err error
+				if len(p.Status.ContainerStatuses) != 0 {
+					err = fmt.Errorf("rule command executor is struck in pending state, reason: %v", p.Status.ContainerStatuses[0].State.Waiting.Reason)
+				} else {
+					err = fmt.Errorf("rule command executor is struck in pending state")
+				}
+				logrus.Errorf("%v", err)
+				return err
```

Comment:
missing pod delete before return?

Reply:
There is a defer func that takes care of deleting the pod.